File: mockmate-backend/src/main/resources/runners/java/runner_linked_list.java
import java.util.*;

public class Main {

    static class ListNode {
        int val;
        ListNode next;

        ListNode(int val) {
            this.val = val;
        }
    }

    static ListNode buildList(String input) {
        if (input.isEmpty())
            return null;
        String[] parts = input.trim().split("\\s+");
        if (parts.length == 0 || parts[0].isEmpty())
            return null;
        ListNode dummy = new ListNode(0);
        ListNode curr = dummy;
        for (String p : parts) {
            curr.next = new ListNode(Integer.parseInt(p));
            curr = curr.next;
        }
        return dummy.next;
    }

    static String serializeList(ListNode head) {
        StringBuilder sb = new StringBuilder();
        while (head != null) {
            if (sb.length() > 0)
                sb.append(" ");
            sb.append(head.val);
            head = head.next;
        }
        return sb.toString();
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String input = sc.hasNextLine() ? sc.nextLine().trim() : "";
        ListNode head = buildList(input);

        Solution sol = new Solution();
        Object result = sol.{{methodSignature}}(head);

        if (result instanceof ListNode) {
            System.out.println(serializeList((ListNode) result));
        } else {
            System.out.println(result);
        }
    }
}

{{USER_CODE}}

File: mockmate-backend/src/main/resources/runners/python/runner_linked_list.py
import sys

class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next

def build_list(values):
    dummy = ListNode(0)
    curr = dummy
    for v in values:
        if v != 'null':
            curr.next = ListNode(int(v))
            curr = curr.next
    return dummy.next

def serialize_list(head):
    result = []
    while head:
        result.append(str(head.val))
        head = head.next
    return ' '.join(result)

{{USER_CODE}}

lines = sys.stdin.read().strip().split('\n')
values = lines[0].strip().split()
head = build_list(values)
sol = Solution()
result = sol.{{methodSignature}}(head)
if isinstance(result, ListNode):
    print(serialize_list(result))
elif isinstance(result, bool):
    print(str(result).lower())
else:
    print(result)
